collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FormHierarchyTest.java

```diff
@@ -170,4 +172,27 @@ public void deletingLastGroupAndAddingOneShouldNotBreakHierarchy() {
                 .deleteGroup()
                 .addGroup();
     }
+
+    @Test
+    //https://github.com/getodk/collect/issues/4570
+    public void showRepeatsPickerWhenFirstRepeatIsEmpty() {
+        new MainMenuPage()
+                .startBlankFormWithRepeatGroup("Empty First Repeat", "Repeat")
+                .clickOnAdd(new FormEntryPage("Empty First Repeat"))
+                .answerQuestion("Question in repeat", "Not empty!")
+                .clickGoToArrow()
+                .clickGoUpIcon()
+                .clickGoUpIcon()
+                .assertText("Repeat", "Repeatable Group");
+    }
+
+    @Test
+    public void empty_first_repeat() {
```

Comment:
I don't think we need both tests now. As far as I can see `showRepeatsPickerWhenFirstRepeatIsEmpty()` enforces the correct code on its own.

Reply:
Definitely, I only put the earlier one back as a check.